packages/foundry-sdk-generator/src/generate/betaClient/minifyBundle.ts

```diff
@@ -150,16 +223,47 @@ export class ProjectMinifier {
               const importsFromNamespace = new Set(
                 exportSourceFile.getExportedDeclarations().keys(),
               );
-              this.pushNextVisit(exportSourceFile, importsFromNamespace);
+              this.pushNextVisit(
+                exportSourceFile,
+                importsFromNamespace,
+                "minifyProjectA",
+              );
             }
           }
         }
 
         const namedExports = declaration.getNamedExports();
 
+        function deleteFromSet(set: Set<string>, toDelete: Set<string>) {
+          for (const d of toDelete) {
+            set.delete(d);
+          }
+        }
+
         // export * from "..."
         if (namedExports.length === 0) {
-          this.pushNextVisit(exportSourceFile, visitImports.imports);
+          const depExportNames = new Set(
+            exportSourceFile.getExportSymbols().map(s => s.getName()).filter(
```

Comment:
If the file we are looking at has `export * from "otherFile"` but we are instructed to only keep `fooBar` from "this file", then we only want `fooBar` from "otherFile".